lib/opam.ml

```diff
@@ -238,3 +238,29 @@ module Pos = struct
     let pos = from_value value in
     errorf ~pos fmt
 end
+
+module Value = struct
+  module String = struct
+    let from_value value =
+      match (value : OpamParserTypes.FullPos.value) with
+      | { pelem = String s; _ } -> Ok s
+      | { pos; _ } -> Pos.errorf ~pos "Expected a string"
+
+    let to_value s = Pos.with_default (OpamParserTypes.FullPos.String s)
+  end
+
+  module List = struct
+    let from_value ~elm_from_value value =
```

Comment:
Maybe calling it `element` or `item` could be a better naming? I don't know, there isn't a good convention I am aware of, it's just that `elm_from_value` sounds oddly complicated for what is essentially `~f` in `List.map`. Maybe `~f` could actually be a good alternative since it takes the same `'a` as `from_value` takes a `'a list` of?

Reply:
I think `f` is a bit to generic here considering we need a very specific kind of function in that case, as opposed to `List.map`. I believe having a good argument name will make it easier to read code using those combinator-like functions.

I'm happy to change it to `element` as `elm` might be less clear. I went for `elm` as it's used in `Set.S` from the standard library and is slightly shorter than `element` which is fairly consistently used across the standard library documentation for the `List` module.

Reply:
Maybe just `~elm` in such case, to match the Stdlib?

Reply:
I'm a bit confused now, I don't really understand why `elm` would be an improvement over what we currently have.
Could you explain what confused or bothered you with `elm_from_value` in the first place? I think this would help me fix this naming problem.

I can try better explaining why I named it like this as well:
To clarify, in the std lib, `elm` is the type name for the elements of a `Set.t`, and the term `element` is used in the documentation of list to describe elements of a list. It's never used as a labelled argument directly or at least not that I know of.

Here we expect a function for this argument, a function that converts opam values. This function will be applied to all elements of a list to convert an opam value list into an ocaml list, hence my original naming. I would find `elm` very confusing as it doesn't help me understand what should be passed here. Again, the interface will likely clarify that but I think it's a good property to make code using those functions readable, without having to jump to the doc or using merlin to show the type of the arguments.

Reply:
I think my confusion is that the argument `elm_from_value` when I first saw it is a rather complex name where I expect something complicated to happen, like some very specific functionality. Then then reading continuing to read the function I realized that the reality is much simpler, this is just the recursive application (as is basically apparent from the signature itself). This makes it in my mind more comparable to `List.map` which applies some named/`f`-named function to its elements or even more comparable, `Fmt.list` which takes an `'a Fmt.t` as an argument where it is pretty clear what it will be used for.

As in a hypothetical alternative
```ocaml
Fmt.list : ?sep:unit Fmt.t -> ~elm_fmt:'a Fmt.t -> 'a list Fmt.t
```
I don't think the signature would any more understandable by giving the argument that gets applied on every element a complex-sounding name.

I don't want to bikeshed this, so if you prefer `elm_from_value` that's fine with me. This is just my explanation why I think a simple name can be good enough even if it doesn't exactly name what it is.

Reply:
ok, I removed the label alltogether then!